### PR TITLE
[PIM-6125] Add validator for family attribute_as_label

### PR DESCRIPTION
**Pull request by pierallard**
**Description (for Contributor and Core Developer)**

Prevent:
- import a family with an attribute_as_label not present in the attributes list
- import a family with an attribute_as_label not identifier nor text type

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | y
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: CHANGELOG-1.7.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - GITHUB-5294: Fixed infinite loading if no attribute is configured as a product identifier, cheers @gplanchat!
 - GITHUB-5337: Fixed Widget Registry. Priority is now taken in account.
 - PIM-6127: In the family import, the attributes required should be in the family
+- PIM-6125: In the family import, the attribute_as_label have to be in the family and its type has to be identifier or text
```
Comment (jjanvier) [inline CHANGELOG-1.7.md]: has

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        }
+    }
+
+    /**
+     * @param FamilyInterface $family
+     *
+     * @return bool
+     */
+    protected function isAttributeAsLabelPresent(FamilyInterface $family)
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: isAttributeAsLabelExist ? or isset maybe ?

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FamilyAttributeAsLabelValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($family, Constraint $constraint)
+    {
+        if ($family instanceof FamilyInterface) {
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: early return please

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    /**
+     * @param FamilyInterface $family
+     *
+     * @return bool
+     */
+    protected function isAttributeAsLabelPresent(FamilyInterface $family)
+    {
+        $attributeAsLabelCode = $family->getAttributeAsLabel()->getCode();
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: in_array, that's cool you know ^^

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+class FamilyAttributeAsLabel extends Constraint
+{
+    /** @var string */
+    public $messageAttribute = 'The attribute "%attribute%" does not belong to the family, thus it cannot be used as '.
+        'an "attribute as label" for this family';
+
+    /** @var string */
+    public $messageAttributeType = 'Only text and identifier attribute types can be used as "attribute as label" '.
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php]: Only "text" and "identifier" attribute types ?

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+class FamilyAttributeAsLabel extends Constraint
+{
+    /** @var string */
+    public $messageAttribute = "Property 'attribute_as_label' must belong to the family";
+
+    /** @var string */
+    public $messageAttributeType = "Property 'attribute_as_label' only supports 'pim_catalog_text' and ".
```
Comment (grena) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php]: No translations?

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Family attribute_as_label constraint
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
```
Comment (grena) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php]: extra space

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            $this->context->buildViolation($constraint->messageAttributeType)->addViolation();
+        }
+    }
+
+    /**
+     * @param FamilyInterface $family
+     *
+     * @return bool
+     */
+    protected function isSetAttributeAsLabel(FamilyInterface $family)
```
Comment (grena) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: I don't understand the naming. What do you think of `doesAttributeAsLabelBelongToFamily` ?

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php (hunk 6 above)
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabel.php]: There is no translation on validation for now

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php (hunk 8 above)
Comment (pierallard) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: IRL

File: src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php (hunk 2 above)
Comment (a2xchip) [inline src/Pim/Component/Catalog/Validator/Constraints/FamilyAttributeAsLabelValidator.php]: *Exists

**Pull request merged**